src/blocks/scratch3_procedures.js

```diff
@@ -1,44 +1,46 @@
-var Scratch3ProcedureBlocks = function (runtime) {
+class Scratch3ProcedureBlocks {
+    constructor (runtime) {
+        /**
+         * The runtime instantiating this block package.
+         * @type {Runtime}
+         */
+        this.runtime = runtime;
+    }
+
     /**
-     * The runtime instantiating this block package.
-     * @type {Runtime}
+     * Retrieve the block primitives implemented by this package.
+     * @return {object.<string, Function>} Mapping of opcode to Function.
      */
-    this.runtime = runtime;
-};
-
-/**
- * Retrieve the block primitives implemented by this package.
- * @return {object.<string, Function>} Mapping of opcode to Function.
- */
-Scratch3ProcedureBlocks.prototype.getPrimitives = function () {
-    return {
-        procedures_defnoreturn: this.defNoReturn,
-        procedures_callnoreturn: this.callNoReturn,
-        procedures_param: this.param
-    };
-};
+    getPrimitives () {
+        return {
+            procedures_defnoreturn: this.defNoReturn,
+            procedures_callnoreturn: this.callNoReturn,
+            procedures_param: this.param
+        };
+    }
 
-Scratch3ProcedureBlocks.prototype.defNoReturn = function () {
-    // No-op: execute the blocks.
-};
+    defNoReturn () {
+        // No-op: execute the blocks.
+    }
 
-Scratch3ProcedureBlocks.prototype.callNoReturn = function (args, util) {
-    if (!util.stackFrame.executed) {
-        var procedureCode = args.mutation.proccode;
-        var paramNames = util.getProcedureParamNames(procedureCode);
-        for (var i = 0; i < paramNames.length; i++) {
-            if (args.hasOwnProperty('input' + i)) {
-                util.pushParam(paramNames[i], args['input' + i]);
+    callNoReturn (args, util) {
+        if (!util.stackFrame.executed) {
+            const procedureCode = args.mutation.proccode;
+            const paramNames = util.getProcedureParamNames(procedureCode);
+            for (let i = 0; i < paramNames.length; i++) {
+                if (args.hasOwnProperty(`input${i}`)) {
```

Comment:
This is one of my favorite ES6 features: more readable AND (possibly) more efficient!